cpp/cmake/Modules/ConfigureArrow.cmake

```diff
@@ -47,48 +66,33 @@ else()
     set(PARALLEL_BUILD --parallel)
     message("STATUS Enabling Parallel CMake build")
 endif($ENV{TRAVIS})
-
+ 
 execute_process(
-    COMMAND ${CMAKE_COMMAND} --build ${PARALLEL_BUILD} ..
+    COMMAND ${CMAKE_COMMAND} --build ${PARALLEL_BUILD} .
     RESULT_VARIABLE ARROW_BUILD
-    WORKING_DIRECTORY ${ARROW_ROOT}/build)
+    WORKING_DIRECTORY ${ARROW_ROOT}/download)
 
 if(ARROW_BUILD)
-    message(FATAL_ERROR "Building Arrow failed: " ${ARROW_BUILD})
-endif(ARROW_BUILD)
-
-set(ARROW_GENERATED_IPC_DIR 
-    "${ARROW_ROOT}/build/src/arrow/ipc")
-
-configure_file(${ARROW_GENERATED_IPC_DIR}/File_generated.h ${CMAKE_SOURCE_DIR}/include/cudf/ipc_generated/File_generated.h COPYONLY)
-configure_file(${ARROW_GENERATED_IPC_DIR}/Message_generated.h ${CMAKE_SOURCE_DIR}/include/cudf/ipc_generated/Message_generated.h COPYONLY)
-configure_file(${ARROW_GENERATED_IPC_DIR}/Schema_generated.h ${CMAKE_SOURCE_DIR}/include/cudf/ipc_generated/Schema_generated.h COPYONLY)
-configure_file(${ARROW_GENERATED_IPC_DIR}/Tensor_generated.h ${CMAKE_SOURCE_DIR}/include/cudf/ipc_generated/Tensor_generated.h COPYONLY)
+    message(FATAL_ERROR "Build step for Arrow failed: ${ARROW_BUILD}")
+endif()
```

Comment:
Please replace with `endif(ARROW_BUILD)` so that readers know immediately that "this is the end of the if-block for  building Arrow"

Reply:
Understood and corrected